### ISSUE TITLE
[feature request] Analysing licenses of dependencies

### ISSUE THREAD

**Issue by svdo** [chaoss/grimoirelab-graal#81]
# TL/DR

I think it would be valuable for Graal to be able to analyse the (open source) software licenses of _dependencies_ of my software.

# Rationale

People and organisations generally want to adhere to the licenses under which software is made available to them. Nowadays this has become so difficult that tooling is needed to help. Take for example the Javascript/nodejs ecosystem. When you develop even a simple client-server application, you easily end up using hundreds and hundreds of open source packages, either directly or through transitive dependencies. It's virtually impossible to collect and verify all this manually for every release of every product.

# Feature Request

I envision a component of Graal that can create a "bill of materials" of my software. A table containing _all_ dependencies of my software, both directly and indirectly. The table should have columns for:

- Name of the dependency
- Where the dependency can be found (could be a link to a package manager repository such as maven / npm / etc; or a link to the source code archive)
- Software license of the dependency; normalised somehow to easily search and filter

Furthermore, a view where the dependency _tree_ is visualised seems useful, but that is probably more of a V2 feature / nice-to-have.

# Note

Please note that this probably depends on #80 to be a useful feature.

Thanks for your consideration and this great software!
